### PR TITLE
Set message content type/encoding, consistently print errors, enable amqp, and IoT Edge connection

### PR DESCRIPTION
**Pull request by drwill-ms**
If message ContentEncoding and ContentType aren't set, then message routing based on a body property won't work in IoT Hub. We should consistently set these in our samples.

I also ran into an error, but the sample didn't print what that error was. I found all cases of encountering errors and made sure we printed them in a consistent way.

### CODE REVIEW

**Comment by drwill-ms** [comment]
I've tested all combos (MQTT | AMQP) x (IoT Hub | IoT Edge)

File: README.md [hunk 1]
```diff
@@ -5,20 +5,25 @@ author: shizn
 ---
 
 # IoT Hub Raspberry Pi 3 Client application
+
```
Comment (drwill-ms) [inline README.md]: Mostly just fixes based on MD Lint in here.

File: README.md [hunk 2]
```diff
@@ -33,34 +38,68 @@ sudo apt-get -y install nodejs
...
 ### Send Cloud-to-Device command
+
 You can send a C2D message to your device. You can see the device prints out the message and blinks once receiving the message.
 
 ### Send Device Method command
-You can send `start` or `stop` device method command to your Pi to start/stop sending message to your IoT hub.
+
+You can send `start` or `stop` device method command to your Pi to start/stop sending message to your IoT Hub.
+
+### Run the client using the AMQP transport
```
Comment (drwill-ms) [inline README.md]: From here on, for new functionality I added, and some troubleshooting steps that worked for me.

**Comment by shizn** [comment]
@yzhong94 Could you or your team help review this?

File: index.js [hunk 3]
```diff
@@ -100,6 +119,21 @@ function initClient(connectionStringParam, credentialPath) {
 
     console.log('[Device] Using X.509 client certificate authentication');
   }
+
+  if (connectionString.GatewayHostName && config.iotEdgeRootCertFilePath) {
```
Comment (drwill-ms) [inline index.js]: This makes it work with an IoT Edge gateway, when it has generated (dev) certs on it.

File: package.json [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
+  "description": "Client application for Raspberry Pi collect temperature and humidity data from BME280, then interact with Azure IoT Hub",
   "main": "index.js",
   "scripts": {
     "test": "node_modules/.bin/eslint *.js"
   },
   "dependencies": {
     "az-iot-bi": "^0.1.18",
-    "azure-iot-device": "1.1.7",
-    "azure-iot-device-mqtt": "1.1.7",
+    "azure-iot-device": "^1.9.3",
```
Comment (drwill-ms) [inline package.json]: This was way out of date. The tutorial hit several bugs which have since been fixed. We should take the latest version.

**Pull request merged**